modules/apps/headless/headless-batch-engine/headless-batch-engine-impl/src/main/java/com/liferay/headless/batch/engine/internal/resource/v1_0/ImportTaskResourceImpl.java

```diff
@@ -419,8 +435,17 @@ private ImportTask _importFile(
 				binaryFile.getInputStream());
 		}
 		else {
+			String fileName = binaryFile.getFileName();
+
+			String fileExtension = _file.getExtension(fileName);
+
+			String trimmedFileName = fileName.replaceAll(
+				"-.*.", "." + fileExtension);
```

Comment:
this is not good. Please try this with file name:
file-name-with-many-dashes-to-replace-randomXyZ.csv

please instead regex use:
fileName.substring(0, fileName.lastIndexOf("-")) + "." +fileExtension